extensions/wikia/AdEngine/js/slot/premiumFloatingMedrec.js

```diff
@@ -0,0 +1,133 @@
+/*global define*/
+define('ext.wikia.adEngine.slot.premiumFloatingMedrec', [
+	'ext.wikia.adEngine.adContext',
+	'ext.wikia.adEngine.slot.service.viewabilityHandler',
+	'wikia.document',
+	'wikia.log',
+	'wikia.throttle',
+	'wikia.window'
+], function (
+	adContext,
+	viewabilityHandler,
+	doc,
+	log,
+	throttle,
+	win
+) {
+	'use strict';
+	var logGroup = 'ext.wikia.adEngine.slot.premiumFloatingMedrec';
+
+	function init() {
+		var adSlot = doc.createElement('div'),
+			context = adContext.getContext(),
+			maxChanges = 6,
+			onScroll,
+			placeHolder = doc.getElementById('WikiaAdInContentPlaceHolder'),
+			recirculation = doc.getElementById('recirculation-rail'),
+			refresh = {
+				refreshAdPos: 0,
+				lastRefreshTime: new Date(),
+				refreshNumber: 0,
+				adVisible: true,
+				heightScrolledThreshold: 10,
+				refreshDelay: 10000
+			},
+			slotName = 'INCONTENT_BOXAD_1',
+			startPosition = placeHolder.offsetTop,
+			swapRecirculationAndAd;
+
+		adSlot.className = 'wikia-ad';
+		adSlot.setAttribute('id', slotName);
+
+		function shouldSwitchModules(currentHeightPosition) {
+			var heightScrolled = Math.abs(currentHeightPosition - refresh.refreshAdPos),
+				timeDifference = (new Date()) - refresh.lastRefreshTime,
+				result = heightScrolled > refresh.heightScrolledThreshold &&
+					timeDifference > refresh.refreshDelay &&
+					refresh.refreshNumber < maxChanges;
+console.log(timeDifference);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this logging statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2228)